### PR TITLE
Add support for the feedback API

### PR DESCRIPTION
**Pull request by ankrgyl**
This change extends the SDK to support the feedback API, documented at https://www.braintrustdata.com/docs/guides/logging#user-feedback.

There are two kinds of feedback you can provide:
* The `logFeedback` / `log_feedback` method can be run outside of the context of a span, and given an `id` + object ids, will add extra information to a span as a tracked auditable update.
* We also now support creating subspans of older requests, simply by specifying the object ids and request id. This relies on new functionality in the API that swaps the `_parent_id` field with the appropriate `root_span_id` and `span_parents`.

### CODE REVIEW

**Review by manugoyal — APPROVED** [review]
lgtm! Didn't look carefully at the python code. LMK if there's anything substantially different from JS worth looking at.

File: js/src/logger.ts [hunk 1]
```diff
@@ -1722,6 +1882,9 @@ export class SpanImpl implements Span {
     if (args.parentSpanInfo) {
       this.internalData.span_parents = [args.parentSpanInfo.span_id];
     }
+    if (!isEmpty(args.parentId)) {
```
[the remark this answers is not in the document]
Comment (manugoyal) [reply js/src/logger.ts:1885]: On second thought, if we make this a union against `parentSpanInfo`, then maybe doing it in an if/else branch makes sense.

File: js/src/logger.ts [hunk 2]
```diff
@@ -401,6 +422,82 @@ export interface LogOptions<IsAsyncFlush> {
...
+      return {
+        id,
+        ...updateEvent,
+        ...(await trueParentIds),
+        [AUDIT_SOURCE_FIELD]: source,
+        [AUDIT_METADATA_FIELD]: metadata,
+        [IS_MERGE_FIELD]: true,
+      };
+    })();
+    bgLogger.log([record]);
```
[the remark this answers is not in the document]
Comment (ankrgyl) [reply js/src/logger.ts:476]: good idea

File: js/src/logger.ts [hunk 3]
```diff
@@ -586,6 +721,7 @@ type ExperimentEvent = Partial<InputField> &
     created: string;
     span_parents: string[];
     span_attributes: Record<string, unknown>;
+    [PARENT_ID_FIELD]: string;
```
[the remark this answers is not in the document]
Comment (ankrgyl) [reply js/src/logger.ts:724]: fixed

File: js/src/logger.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ankrgyl) [reply js/src/logger.ts:1885]: Yeah fixed

**Pull request merged**